core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -346,19 +349,15 @@ private InputStream getUncompressedBody() {
   }
 
   private InputStream getCompressedBody() {
-    if (compression == Compression.NONE) {
+    if (decompressor == MessageEncoding.NONE) {
```

Comment:
Using a sentinel object has some benefits over using null.  The best one IMO is being easy to find references to it.  It also buys a little extra meaning in the context of nullable things, notably an explicitly "empty" thing, vs an implicitly empty thing, such as null.  It also provides a a hook to providing clarity in the error message ("You should not have called me")  vs getting an NPE.  This code does such as check, but the call site has to make it rather than putting that logic in the object itself.


On a tangential note, I could feasibly see the error below being caused by the remote side (not setting any compression, and then sending a compressed message).